### PR TITLE
Deprecates Source in JobMappingTransformer config

### PR DESCRIPTION
**Pull request by nickzelei**

### CODE REVIEW

File: frontend/apps/web/components/jobs/SchemaTable/RowAlert.tsx [hunk 1]
```diff
@@ -5,24 +5,21 @@ import {
...
-import { Row } from '@tanstack/react-table';
 import { ReactElement } from 'react';
-import { SchemaConstraintHandler } from './schema-constraint-handler';
+import {
+  ColumnKey,
+  SchemaConstraintHandler,
+} from './schema-constraint-handler';
 
 interface Props {
-  row: Row<{ schema: string; table: string; column: string }>;
```
Comment (nickzelei) [inline frontend/apps/web/components/jobs/SchemaTable/RowAlert.tsx:13]: TS was continually unhappy about this even though it builds. This update just makes it less type dependent on the Row type and pushes it to the caller.

File: frontend/apps/web/components/jobs/SchemaTable/SchemaColumns.tsx [hunk 2]
```diff
@@ -127,10 +127,15 @@ export function getSchemaColumns(props: Props): ColumnDef<RowData>[] {
...
+        const columnKey: ColumnKey = {
+          schema: row.getValue<string>('schema'),
+          table: row.getValue<string>('table'),
+          column: row.getValue<string>('column'),
+        };
         return (
           <div className="flex flex-row gap-2 items-center">
             <SchemaRowAlert
-              row={row}
+              rowKey={columnKey}
```
Comment (nickzelei) [inline frontend/apps/web/components/jobs/SchemaTable/SchemaColumns.tsx:138]: much cleaner, doesn't expose the `Row` anymore to this component

File: worker/pkg/benthos/default_transform/processor.go [hunk 3]
```diff
@@ -219,9 +219,18 @@ func initDefaultTransformers(defaultTransformerMap map[primitiveType]*mgmtv1alph
...
+	}
+
+	source :=
+		t.GetSource() != mgmtv1alpha1.TransformerSource_TRANSFORMER_SOURCE_UNSPECIFIED &&
+			t.GetSource() != mgmtv1alpha1.TransformerSource_TRANSFORMER_SOURCE_PASSTHROUGH
+
+	config := t.GetConfig().GetConfig()
+	passthrough := t.GetConfig().GetPassthroughConfig()
+	// we only want to process if there is a non-nil config that is not passthrough
+	return source || (config != nil && passthrough == nil)
```
Comment (alishakawaguchi) [inline worker/pkg/benthos/default_transform/processor.go:233]: can this also be updated to be a switch?

**Review by alishakawaguchi — APPROVED** [review]
🏅 
left one comment

File: worker/pkg/benthos/default_transform/processor.go (hunk 3 above)
Comment (nickzelei) [reply worker/pkg/benthos/default_transform/processor.go:233]: ah yea good catch. I think this function is duplicated in two spots.

**Pull request merged**